core/controller/src/main/scala/whisk/core/controller/Authenticate.scala

```diff
@@ -64,17 +64,18 @@ trait Authenticate extends Logging {
                         info(this, s"authentication not valid")
                         None
                     }
+                } recover {
+                    case _: NoDocumentException | _: IllegalArgumentException =>
+                        info(this, s"authentication not valid")
+                        None
+                    case t: Throwable =>
+                        error(this, s"authentication error: $t")
+                        None
```

Comment:
This is not semantically preserving i believe. In this case you `recover` any error to a `None` whereas before, only `NoDocumentException` and `IllegalArgumentException` have been recovered. Was this change intended?